### ISSUE TITLE
Grpc traffic broken, shows HTTP/2.0 on client proxy and HTTP/1.1 on server proxy

### ISSUE THREAD

**Issue by Sovietaced** [istio/istio#18854]
**Bug description**

I have two services which speak grpc. When the request hits the client proxy it is logged as HTTP/2.0 but when the request hits the server proxy it is logged as HTTP/1.1. Consequently when the response is parsed by the client a parsing related exception is thrown. What is going on here?

Client Proxy
`
[2019-11-11T15:40:03.862Z] "POST /ai.argo.viaduct.proto.v1.ViaductAPI/Query HTTP/2" 200 - "-" "-" 1241 1280 405 405 "-" "grpc-java-netty/1.21.1" "2ef9594f-c22b-9e49-8d87-de76783f5439" "192.168.192.5:8085" "192.168.192.5:8085" PassthroughCluster - 192.168.192.5:8085 192.168.192.11:50252 -
`

Server Proxy

`[2019-11-11T15:40:03.863Z] "POST /ai.argo.viaduct.proto.v1.ViaductAPI/Query HTTP/1.1" 200 - "-" "-" 1241 1280 405 404 "-" "grpc-java-netty/1.21.1" "2ef9594f-c22b-9e49-8d87-de76783f5439" "192.168.192.5:8085" "127.0.0.1:8085" inbound|8085|grpc-viaduct|viaduct.overwatch-development.svc.cluster.local - 192.168.192.5:8085 192.168.192.11:50230 -`

Service Description

    Name:                     viaduct
    Namespace:                overwatch-development
    Labels:                   <none>
    Annotations:              external-dns.alpha.kubernetes.io/hostname: viaduct.dev.overwatch.argo.tools.
                              kubectl.kubernetes.io/last-applied-configuration:
                                {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{"external-dns.alpha.kubernetes.io/hostname":"viaduct.dev.overwatch.argo.too...
                              service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
                              service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
    Selector:                 app=viaduct
    Type:                     LoadBalancer
    IP:                       172.20.86.120
    LoadBalancer Ingress:     internal-a00aa67cec47111e9ac750a2cc3eb559-1876607380.us-east-1.elb.amazonaws.com
    Port:                     tcp  8914/TCP
    TargetPort:               rs/TCP
    NodePort:                 tcp  31717/TCP
    Endpoints:                192.168.128.3:8914,192.168.192.5:8914
    Port:                     tls  443/TCP
    TargetPort:               secure-rs/TCP
    NodePort:                 tls  31490/TCP
    Endpoints:                192.168.128.3:443,192.168.192.5:443
    Port:                     grpc-viaduct  8085/TCP
    TargetPort:               grpc/TCP
    NodePort:                 grpc-viaduct  30266/TCP
    Endpoints:                192.168.128.3:8085,192.168.192.5:8085
    Session Affinity:         None
    External Traffic Policy:  Cluster
    Events:
      Type    Reason                Age                 From                Message
      ----    ------                ----                ----                -------
      Normal  EnsuringLoadBalancer  16m (x22 over 30d)  service-controller  Ensuring load balancer
      Normal  EnsuredLoadBalancer   16m (x22 over 30d)  service-controller  Ensured load balancer

**Expected behavior**

grpc traffic shows at HTTP/2.0 on server proxy and grpc traffic not broken

**Steps to reproduce the bug**

Setup client/server which use grpc and have client send traffic directly to service endpoint (pod IP:PORT). 

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.10-eks-825e5d", GitCommit:"825e5de08cb05714f9b224cd6c47d9514df1d1a7", GitTreeState:"clean", BuildDate:"2019-08-18T03:58:32Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

```
jparraga@jparraga-5820:~/workspace/cluster_eks-overwatch-development$ istioctl version --remote
client version: 1.3.1
control plane version: 1.2.2

```

**How was Istio installed?**

Helm

**Environment where bug was observed (cloud vendor, OS, etc)**

AWS EKS

**Comment by Sovietaced** [comment]
It looks like this issue can be avoided with the use of a `ServiceEntry` but that should not be necessary when communicating inside the service mesh.

**Comment by howardjohn** [comment]
This looks like you are sending traffic to `192.168.192.5:8085`. Routing in Istio is based on host header, so `192.168.192.5:8085` would not be treated as "communicating inside the mesh". So I think this is expected that ServiceEntry would be needed here

**Comment by Sovietaced** [comment]
> This looks like you are sending traffic to `192.168.192.5:8085`. Routing in Istio is based on host header, so `192.168.192.5:8085` would not be treated as "communicating inside the mesh". So I think this is expected that ServiceEntry would be needed here

I can provide some more context now... The use case here is that we have service A which wants to make a request to **a specific pod** for service B which holds a sticky, persistent TCP connection to devices located outside the service mesh. 

I'm not sure the best way to handle this use case but I ended up hacking something together that works. I updated service B to use a stateful set which gives me stable hostname identifiers that I can use. Then I setup the following service entry..

```
---
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: viaduct-service-entry
  namespace: <redacted>
spec:
  hosts:
  - "*.viaduct"
  location: MESH_INTERNAL
  ports:
  - name: grpc
    number: 8085
    protocol: GRPC
  resolution: NONE
```

It matches the hostnames of all of the stateful set pods using a wildcard and treats the traffic as mesh internal.

**Comment by howardjohn** [comment]
I am fairly sure this is fixed by https://github.com/istio/istio/issues/24706. Please reopen if not. thanks!

**Comment by Sovietaced** [comment]
> I am fairly sure this is fixed by #24706. Please reopen if not. thanks!

Thanks. I'll give this a shot when I have some time at work.
